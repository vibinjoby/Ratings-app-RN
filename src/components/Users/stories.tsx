import React from 'react'
import { storiesOf } from '@storybook/react-native'
import Users from './component'

storiesOf('Components', module).add('Users', () => (
  <Users
    id={''}
    name={'vibin'}
    typeOfUser={'customer'}
    onDelete={() => console.log('delete clicked')}
  />
))
