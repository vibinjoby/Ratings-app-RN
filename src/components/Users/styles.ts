import { StyleSheet } from 'react-native'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  userPic: {
    width: 45,
    height: 45,
    borderRadius: 22.5,
  },
  nameContainer: {
    marginLeft: 10,
  },
  name: {
    ...Typography.BoldFont,
    fontSize: 16,
  },
  userType: {
    ...Typography.Caption1.regular,
    fontSize: 12,
  },
  deleteIc: {
    position: 'absolute',
    right: 0,
    width: 24,
    height: 24,
  },
})
