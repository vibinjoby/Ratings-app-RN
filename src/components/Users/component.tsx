import React from 'react'
import { Image, Text, TouchableOpacity, View } from 'react-native'

import styles from './styles'
import commons from '../../configs/commonConst'

export interface UsersProps {
  id: string
  name: string
  typeOfUser: string
  onDelete: (arg0: string) => void
  index?: number
}

const Users: React.FC<UsersProps> = ({ id, name, typeOfUser, onDelete, index }: UsersProps) => {
  return (
    <View style={styles.container}>
      <Image style={styles.userPic} source={{ uri: commons.DUMMY_PIC }} />
      <View style={styles.nameContainer}>
        <Text testID="name" style={styles.name}>
          {name}
        </Text>
        <Text testID="userType" style={styles.userType}>
          {typeOfUser}
        </Text>
      </View>
      <TouchableOpacity
        testID={`deleteIc${index}`}
        activeOpacity={0.9}
        onPress={() => onDelete(id)}
        style={styles.deleteIc}
      >
        <Image source={require('../../assets/deleteIc/deleteIc.png')} />
      </TouchableOpacity>
    </View>
  )
}

export default Users
