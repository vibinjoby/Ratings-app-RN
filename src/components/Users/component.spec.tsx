import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import Users from './component'

describe('Users', () => {
  const MOCK_FN = jest.fn()
  const MOCK_ID = '123'
  const MOCK_NAME = 'name'
  const MOCK_USER_TYPE = 'usertype'

  it('should render', () => {
    const { toJSON } = render(
      <Users id={MOCK_ID} name={MOCK_NAME} typeOfUser={MOCK_USER_TYPE} onDelete={MOCK_FN} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should display all values and should be deletable', () => {
    const { getByTestId } = render(
      <Users
        id={MOCK_ID}
        index={0}
        name={MOCK_NAME}
        typeOfUser={MOCK_USER_TYPE}
        onDelete={MOCK_FN}
      />,
    )
    expect(getByTestId('name').props.children).toBe(MOCK_NAME)
    expect(getByTestId('userType').props.children).toBe(MOCK_USER_TYPE)
    fireEvent.press(getByTestId('deleteIc0'))
    expect(MOCK_FN).toBeCalledWith(MOCK_ID)
  })
})
