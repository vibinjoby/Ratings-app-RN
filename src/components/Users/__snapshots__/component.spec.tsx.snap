// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://source.unsplash.com/user/picoftasty",
      }
    }
    style={
      Object {
        "borderRadius": 22.5,
        "height": 45,
        "width": 45,
      }
    }
  />
  <View
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Poppins-Bold",
          "fontSize": 16,
        }
      }
      testID="name"
    >
      name
    </Text>
    <Text
      style={
        Object {
          "color": "#7C7C7C",
          "fontFamily": "Poppins-Regular",
          "fontSize": 12,
        }
      }
      testID="userType"
    >
      usertype
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 24,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "width": 24,
      }
    }
    testID="deleteIc"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/deleteIc/deleteIc.png",
        }
      }
    />
  </View>
</View>
`;
